Dapper/SqlMapper.DapperRow.Descriptor.cs

```diff
@@ -98,7 +98,7 @@ public RowBoundPropertyDescriptor(Type type, string name, int index) : base(name
                 public override object GetValue(object component)
                     => ((DapperRow)component).TryGetValue(_index, out var val) ? (val ?? DBNull.Value): DBNull.Value;
                 public override void SetValue(object component, object value)
-                    => ((DapperRow)component).SetValue(_index, value is DBNull ? null : value);
+                    => ((DapperRow)component).SetValue(_index, value is DBNull ? null! : value);

```

Comment:
Should DapperRow accept a null? The whole nullability of DapperRow seems to represent not-nullable, but we can fail in places...I think it's really object? rather than object? in DapperRow (and we can remove the `!` here)